### PR TITLE
Update dependencies 2024-05-16

### PR DESCRIPTION
**Pull request by Nicoretti**
- **Update dependencies**
- **Update github workflows**
- **Update changelog**

### Submission Checklist

* [x] Bumped version number
* [x] Updated Documentation
* [x] Updated API Documentation
* [x] Updated the changelog

### CODE REVIEW

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+
+  cd-job:
+    name: Continues Delivery
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: SCM Checkout
+        uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (Nicoretti) [reply .github/workflows/build-and-publish.yml:17]: Sadly this is not as trivial: https://github.com/exasol/python-toolbox/issues/181

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  tests-job:
+    name: Tests (Python-${{ matrix.python-version }}, Exasol-${{ matrix.exasol-version}})
+    needs: [build-documentation-job, lint-job, type-check-job]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        exasol-version: ["7.1.9"]
```
[the remark this answers is not in the document]
Comment (Nicoretti) [reply .github/workflows/checks.yml:89]: One needs to check the used itde version if thats possible. Still see the point but it also should a task/issue on it's own.

File: .github/workflows/build-and-publish.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaklakariada) [reply .github/workflows/build-and-publish.yml:17]: The ticket mentions `actions/upload-artifact`, this is about `actions/checkout`. But if that's a problem it's fine.

[the remark this answers is not in the document]
Comment (Nicoretti) [reply .github/workflows/build-and-publish.yml:17]: good point, my bad ... context switching gone wrong :sweat_smile:

File: .github/workflows/checks.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nicoretti) [reply .github/workflows/checks.yml:95]: This should be fixed in the `exasol-toolbox` which is providing those workflows, I've created an issue for that, see https://github.com/exasol/python-toolbox/issues/184

File: .github/workflows/build-and-publish.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  workflow_call:
+    secrets:
+      PYPI_TOKEN:
+          required: true
+
+jobs:
+
+  cd-job:
+    name: Continues Delivery
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (Nicoretti) [reply .github/workflows/build-and-publish.yml:13]: This should be fixed in the `exasol-toolbox` which is providing those workflows, I've created an issue for that, see https://github.com/exasol/python-toolbox/issues/184

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**